kolibri/core/content/management/commands/generate_fixtures.py

```diff
@@ -0,0 +1,516 @@
+import random
+
+from django.apps import apps
+from django.conf import settings
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.db import connections
+from le_utils.constants import content_kinds
+from le_utils.constants import file_formats
+from le_utils.constants import format_presets
+from le_utils.constants import languages
+from le_utils.constants.labels import learning_activities
+from le_utils.constants.labels.accessibility_categories import (
+    ACCESSIBILITYCATEGORIESLIST,
+)
+from le_utils.constants.labels.levels import LEVELSLIST
+from le_utils.constants.labels.needs import NEEDSLIST
+from le_utils.constants.labels.subjects import SUBJECTSLIST
+
+from kolibri.core.content.apps import KolibriContentConfig
+from kolibri.core.content.models import *
+
+
+class Command(BaseCommand):
+
+    help = "generate fixtures for the specified app"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--app",
+            type=str,
+            default="content",
+            help="app name of which we will generate fixtures for its models",
+        )
+        parser.add_argument(
+            "--file_name",
+            type=str,
+            default="new_fixtures",
+            help="fixtures file name and path",
+        )
+        parser.add_argument(
+            "--mode",
+            type=str,
+            default="default_db",
+            help="options 'build_fixtures', 'default_db' ",
+        )
+        parser.add_argument(
+            "--levels", type=int, default=2, help="number of tree levels"
+        )
+
+    def switch_to_memory(self):
+
+        print("\n start switching to memory ....\n\n")
+
+        for db in settings.DATABASES:
+
+            settings.DATABASES[db] = {
+                "ENGINE": "django.db.backends.sqlite3",
+                "NAME": ":memory:",
+            }
+            try:
+                del connections[db]
+            except AttributeError:
+                pass
+            call_command("migrate", interactive=False, database=db)
+
+    def handle(self, *args, **options):
+
+        app_name = options["app"]
+        file_name = options["file_name"]
+        generating_mode = options["mode"]
+        required_levels = options["levels"]
+
+        apps_names = ("content", "lessons", "exams", "logger")
+        modes_option = ("build_fixtures", "default_db")
+
+        if generating_mode not in modes_option:
```

Comment:
You don't need to add this code if you use the right argparse type https://docs.python.org/2.7/library/argparse.html#choices

Reply:
done !